config/cpu-avx2.m4

```diff
@@ -0,0 +1,19 @@
+dnl #
+dnl # Check if CPU supports avx2
+dnl #
+AC_DEFUN([ZFS_AC_CPU_AVX2], [
```

Comment:
The build host is not necessarily the system that uses the modules. This needs to be a runtime check rather than an autotools check.